samples/bluetooth/mesh/light_lc/src/model_handler.c

```diff
@@ -0,0 +1,209 @@
+	k_timer_stop(&ctx->trans_timer);
+	ctx->target_lvl = set->lvl;
+	k_timer_start(&ctx->period_timer,
+		K_MSEC(set->transition->delay),
+		K_MSEC(TRANS_TIMER_POLL_PERIOD_MS));
+	k_timer_start(&ctx->trans_timer,
+		K_MSEC(set->transition->time + set->transition->delay),
+		K_NO_WAIT);
+	printk("New light transition-> Lvl: %d, Time: %d, Delay: %d\n",
+		set->lvl,
+		set->transition->time,
+		set->transition->delay);
+}
+
+static void light_trans_cb(struct k_timer *timer_id)
+{
+	struct lightness_ctx *l_ctx =
+		CONTAINER_OF(timer_id,
+		struct lightness_ctx,
+		period_timer);
+	static s32_t increment_rest;
+	u32_t rem = k_timer_remaining_get(&l_ctx->trans_timer);
+
+	if (rem < TRANS_TIMER_POLL_PERIOD_MS) {
+		l_ctx->current_lvl = l_ctx->target_lvl;
+		k_timer_stop(&l_ctx->period_timer);
+	} else {
+		u32_t itr_left = rem / TRANS_TIMER_POLL_PERIOD_MS;
+		s32_t diff = l_ctx->target_lvl - l_ctx->current_lvl;
+		s32_t scaled_increment =
+			(diff * TRANS_TIMER_SCALE_FACTOR / (s32_t)itr_left) +
+			increment_rest;
+
+		increment_rest = scaled_increment % TRANS_TIMER_SCALE_FACTOR;
+		s32_t actual_increment =
+			scaled_increment /
+			TRANS_TIMER_SCALE_FACTOR;
+
+		l_ctx->current_lvl += actual_increment;
+	}
+	printk("Current light lvl: %d/65535\n", l_ctx->current_lvl);
+	lc_pwm_led_set(l_ctx->current_lvl);
+}
+
+static void light_set(struct bt_mesh_lightness_srv *srv,
+			struct bt_mesh_msg_ctx *ctx,
+			const struct bt_mesh_lightness_set *set,
+			struct bt_mesh_lightness_status *rsp)
+{
+	struct lightness_ctx *l_ctx =
+		CONTAINER_OF(srv,
+			struct lightness_ctx,
+			lightness_srv);
+
+	start_new_light_trans(set, l_ctx);
+	rsp->current = l_ctx->current_lvl;
+	rsp->target = l_ctx->target_lvl;
+	rsp->remaining_time =
+		set->transition->time +
+		set->transition->delay;
+}
+
+static void light_get(struct bt_mesh_lightness_srv *srv,
+			struct bt_mesh_msg_ctx *ctx,
+			struct bt_mesh_lightness_status *rsp)
+{
+	struct lightness_ctx *l_ctx =
+		CONTAINER_OF(srv,
+			struct lightness_ctx,
+			lightness_srv);
+
+	rsp->current = l_ctx->current_lvl;
+	rsp->target = l_ctx->target_lvl;
+	rsp->remaining_time = k_timer_remaining_get(&l_ctx->trans_timer);
+}
+
+struct bt_mesh_lightness_srv_handlers lightness_srv_handlers = {
+	.light_set = light_set,
+	.light_get = light_get,
+};
+
+static struct lightness_ctx my_ctx = {
+	.lightness_srv =
+		BT_MESH_LIGHTNESS_SRV_INIT(&lightness_srv_handlers),
+
+};
+
+struct bt_mesh_light_ctrl_srv light_ctrl_srv =
+	BT_MESH_LIGHT_CTRL_SRV_INIT(&my_ctx.lightness_srv);
+
+static struct bt_mesh_elem elements[] = {
+	BT_MESH_ELEM(
+		1, BT_MESH_MODEL_LIST(
+			BT_MESH_MODEL_CFG_SRV(&cfg_srv),
+			BT_MESH_MODEL_HEALTH_SRV(&health_srv, &health_pub)),
+		BT_MESH_MODEL_NONE),
+	BT_MESH_ELEM(
+		2, BT_MESH_MODEL_LIST(
+			BT_MESH_MODEL_LIGHTNESS_SRV(&my_ctx.lightness_srv)),
+		BT_MESH_MODEL_NONE),
+	BT_MESH_ELEM(
+		3, BT_MESH_MODEL_LIST(
+			BT_MESH_MODEL_LIGHT_CTRL_SRV(&light_ctrl_srv)),
+		BT_MESH_MODEL_NONE),
+};
```

Comment:
Compress this to two elements instead (ie move lightness server to elem 1, light_ctrl server to elem 2)

Reply:
Done